### PR TITLE
The inaugural commit for Leeloo's firmware.

### PR DESCRIPTION
**Pull request by ClicketySplit**
## Board/Shield Check-list
 - [Y] This board/shield is tested working on real hardware
 - [Y] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [Y] `.zmk.yml` metadata file added
 - [Y] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [Y] General consistent formatting of DeviceTree files
 - [Y] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [Y] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [Y] If split, no name added for the right/peripheral half
 - [Y] Kconfig.defconfig file correctly wraps *all* configuration in conditional on the shield symbol
 - [Y] `.conf` file has optional extra features commented out
 - [Y] Keyboard/PCB is part of a shipped group buy or is generally available in stock to purchase (OSH/personal projects without general availability should create a zmk-config repo instead)

## Leeloo | Split Keyboard
Leeloo is a 4x6x5m split keyboard that is based on the Lily58 and Corne, which has been [re-]designed from scratch; the schematic, symbols, and footprints have been drawn in KiCad, and the microcontrollers supported include: nice!nano v2, Elite-C, and Pro Micro.

A readme (README.md) file has been included for background information and URLs for more information about Clickety Split.

Leeloo

## Build Results:
Leeloo - Left - Screenshot 2022-03-08 224901

Leeloo - Right - Screenshot 2022-03-08 225016

### CODE REVIEW

File: app/boards/shields/leeloo/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (Nicell) [inline app/boards/shields/leeloo/Kconfig.defconfig:1]:
```suggestion
# Copyright (c) 2022 The ZMK Contributors
```
Please update all copyright headers to _current year_ 😄

File: app/boards/shields/leeloo/boards/nice_nano.overlay [hunk 2]
```diff
@@ -0,0 +1,29 @@
+&spi1 {
```
Comment (Nicell) [inline app/boards/shields/leeloo/boards/nice_nano.overlay:1]: I don't think this file is needed, it doesn't seem that the Leeloo has RGB underglow (which is what this file is for)

File: app/boards/shields/leeloo/leeloo.keymap [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        raise_layer {
+            bindings = <
+&trans     &trans     &trans     &trans     &trans     &trans                           &trans     &trans     &trans     &trans     &trans     &trans
+&trans     &trans     &trans     &trans     &trans     &trans                           &trans     &trans     &trans     &trans     &trans     &trans
+&trans     &trans     &trans     &trans     &trans     &trans                           &trans     &trans     &trans     &trans     &trans     &trans
+&trans     &trans     &trans     &trans     &trans     &trans                           &trans     &trans     &trans     &trans     &trans     &trans
+                      &trans     &trans     &trans     &trans     &trans     &trans     &trans     &trans     &trans     &trans
+            >;
+
+            sensor-bindings = <&inc_dec_kp C_VOL_UP C_VOL_DN>;
```
Comment (Nicell) [inline app/boards/shields/leeloo/leeloo.keymap:54]: I would recommend putting at least some way to interact with BT commands in the default keymap. (BT selection, clear, etc)

File: app/boards/shields/leeloo/leeloo_left.conf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (Nicell) [inline app/boards/shields/leeloo/leeloo_left.conf:1]: Delete this file

File: app/boards/shields/leeloo/leeloo_right.conf [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (Nicell) [inline app/boards/shields/leeloo/leeloo_right.conf:1]: Delete this file, too

**Review by Nicell — CHANGES_REQUESTED** [review]
Excited to see the Leeloo come to fruition! I have a few changes here for you.

File: app/boards/shields/leeloo/leeloo.keymap (hunk 3 above)
Comment (ClicketySplit) [reply app/boards/shields/leeloo/leeloo.keymap]: My goodness!  I completely forgot.  I have BT configuration management in my own zmk-config. LOL!  Thank you!

**Comment by ClicketySplit** [comment]
Hi @Nicell!

Thank you very much for taking the time to review this PR.  I really appreciate your feedback, and hopefully I've implemented your recommendations appropriately.

:)

**Review by Nicell — APPROVED** [review]
@ClicketySplit This PR looks good to me now. We'll merge this once the Leeloo goes up for sale to follow our policy on having every shield merged be readily available commercially (or have a completed GB). Thanks! Feel free to ping here when it's up for sale.

**Comment by ClicketySplit** [comment]
Hi @Nicell!

Wonderful!  **Thank you** for reviewing the alterations!  Absolutely, that makes a lot of sense.  As soon as Leeloo's product page is live, I'll reach out to you.  I've got my fingers cross that Leeloo will be available in about a week and a bit.

:)

**Comment by sabrinafrost** [comment]
I actually purchased a pre-release Leeloo from @ClicketySplit and have been using it for over a week now. Merge approval on a technicality? 😄

**Comment by sabrinafrost** [comment]
@ClicketySplit, small request... would you mind merging main into your branch so I can start using the tap dance changes from https://github.com/zmkfirmware/zmk/pull/1139? I'm working around this with a local repository but it would be great to get my Actions working again.

**Comment by ClicketySplit** [comment]
@sabrinafrost no problem, I've merged and tested within the QA environment and it's 100%.

File: app/boards/shields/leeloo/leeloo.dtsi [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+// | SW7  | SW8  | SW9  | SW10 | SW11 | SW12 |                 | SW12 | SW11 | SW10 | SW9  | SW8  | SW7  |
+// | SW13 | SW14 | SW15 | SW16 | SW17 | SW18 |                 | SW18 | SW17 | SW16 | SW15 | SW14 | SW13 |
+// | SW19 | SW20 | SW21 | SW22 | SW23 | SW24 | SW29 |   | SW29 | SW24 | SW23 | SW22 | SW21 | SW20 | SW19 |
+//                      | SW25 | SW26 | SW27 | SW28 |   | SW28 | SW27 | SW26 | SW25 |
+        map = <
+RC(0,0) RC(0,1) RC(0,2) RC(0,3) RC(0,4) RC(0,5)                 RC(0,6) RC(0,7) RC(0,8) RC(0,9) RC(0,10) RC(0,11)
+RC(1,0) RC(1,1) RC(1,2) RC(1,3) RC(1,4) RC(1,5)                 RC(1,6) RC(1,7) RC(1,8) RC(1,9) RC(1,10) RC(1,11)
+RC(2,0) RC(2,1) RC(2,2) RC(2,3) RC(2,4) RC(2,5)                 RC(2,6) RC(2,7) RC(2,8) RC(2,9) RC(2,10) RC(2,11)
+RC(3,0) RC(3,1) RC(3,2) RC(3,3) RC(3,4) RC(3,5)                 RC(3,6) RC(3,7) RC(3,8) RC(3,9) RC(3,10) RC(3,11)
+                RC(4,1) RC(4,2) RC(4,3) RC(4,4) RC(4,5) RC(4,6) RC(4,7) RC(4,8) RC(4,9) RC(4,10)
```
[the remark this answers is not in the document]
Comment (ClicketySplit) [reply app/boards/shields/leeloo/leeloo.dtsi:28]: Hi @manna-harbour!  Yes, I would agree; I can work on it and submit a PR.

:)

**Pull request merged**